storage/post_policy_v4_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestPostPolicyV4Clone(t *testing.T) {
+	t.Parallel()
+
+	opts := &PostPolicyV4Options{
+		GoogleAccessID: "accessID",
+		PrivateKey:     []byte{},
+		SignBytes: func(b []byte) ([]byte, error) {
+			return b, nil
+		},
+		SignRawBytes: func(bytes []byte) ([]byte, error) {
+			return bytes, nil
+		},
+		Expires:             time.Now(),
+		Style:               VirtualHostedStyle(),
+		Insecure:            true,
+		Fields:              &PolicyV4Fields{ACL: "test-acl"},
+		Conditions:          []PostPolicyV4Condition{},
+		shouldHashSignBytes: true,
+	}
+
+	// Check that all fields are set to a non-zero value, so we can check that
+	// clone accurately clones all fields and catch newly added fields not cloned
+	reflectOpts := reflect.ValueOf(*opts)
+
+	for i := 0; i < reflectOpts.NumField(); i++ {
+		if reflectOpts.Field(i).IsZero() {
```

Comment:
Ooh, apparently this was added in Go 1.13 https://pkg.go.dev/reflect#Value.IsZero